plugins/org.jkiss.dbeaver.ext.hana/src/org/jkiss/dbeaver/ext/hana/model/HANAProcedure.java

```diff
@@ -17,24 +17,169 @@
+                try (JDBCResultSet dbResult = dbStat.executeQuery()) {
+                    while (dbResult.next()) {
+                        String parameterName = dbResult.getString(1);
+                        String columnName = dbResult.getString(2);
+                        String typeName = dbResult.getString(3);
+                        int length = dbResult.getInt(4);
+                        int scale = dbResult.getInt(5);
+                        
+                        List<HANAInplaceTableTypeColumn> inplaceTableType = inplaceTableTypes.get(parameterName);
+                        if (inplaceTableType == null) {
+                            inplaceTableType = new LinkedList<HANAInplaceTableTypeColumn>();
+                            inplaceTableTypes.put(parameterName, inplaceTableType);
+                        }
+                        inplaceTableType.add(new HANAInplaceTableTypeColumn(this, columnName, typeName, inplaceTableType.size() + 1, length, scale));
+                    }
+                }
+            }
+        } catch (SQLException e) {
+            throw new DBException(e, getDataSource());
+        }
+    }
+
+    @Override
+    public void loadProcedureColumns(DBRProgressMonitor monitor) throws DBException {
+        try (JDBCSession session = DBUtils.openMetaSession(monitor, getDataSource(), "Read procedure parameter")) {
+            String stmt;
+            stmt = "SELECT PARAMETER_NAME, DATA_TYPE_NAME, LENGTH, SCALE, POSITION,"+
+                   " TABLE_TYPE_SCHEMA, TABLE_TYPE_NAME, IS_INPLACE_TYPE,"+
+                   " PARAMETER_TYPE, HAS_DEFAULT_VALUE";
+            if (DBSProcedureType.PROCEDURE == getProcedureType()) {
+                stmt += " FROM SYS.PROCEDURE_PARAMETERS"+
+                        " WHERE SCHEMA_NAME=? AND PROCEDURE_NAME=?"+
+                        " ORDER BY POSITION";
+            } else {
+                stmt += " FROM SYS.FUNCTION_PARAMETERS"+
+                        " WHERE SCHEMA_NAME=? AND FUNCTION_NAME=?"+
+                        " ORDER BY POSITION";
+            }
+            try (JDBCPreparedStatement dbStat = session.prepareStatement(stmt)) {
+                dbStat.setString(1, getParentObject().getName());
+                dbStat.setString(2, getName());
+                try (JDBCResultSet dbResult = dbStat.executeQuery()) {
+                    while (dbResult.next()) {
+                        String columnName = dbResult.getString(1);
+                        String typeName = dbResult.getString(2);
+                        int columnSize = dbResult.getInt(3);
+                        int scale = dbResult.getInt(4);
+                        int position = dbResult.getInt(5);
+                        String parameterTypeStr = dbResult.getString(9);
+                        boolean hasInplaceTableType = BOOLEAN_TRUE.equals(dbResult.getString(8));
+                        boolean hasDefaultValue = BOOLEAN_TRUE.equals(dbResult.getString(10));
```

Comment:
Please use `JDBCUtils.safeGetXXX` to silently log the error instead of failing. Boolean comparison can also be replaced with an appropriate utility function.

It also removes the need for having `try-catch` block around it.